### PR TITLE
user-profile-card-redesing

### LINKED ISSUE #2203: Redesign the user info card
- Redesign the user info card to include profile description, location, number of posts supported/supporting.
    - There are some additional features included in the design such as the thanks button but those are for future features and can be ignored for now

### PR DESCRIPTION
**Pull request by aleksandra-b-t**
### What does this pull request aim to achieve? 💯
Here are the changes to the redesign of the user profile, created based on the design on the figma page.
Elements that were supposed to disappear from the profile I did not remove permanently but commented out. In case there will be used in a different place. 
NOTE: The 'Thanks' button is not finished yet, but this is for future features.

_Please be concise and link any related issue(s):_ This PR resolves #2203 

- [✓ ] I have checked that no one else is working on similar changes.
- [✓ ] I have read the latest [**README**](README.md) and understand the development workflow.
- [✓ ] The name of this branch is: **`feature/<branch_name>`** for a regular feature, **`hotfix/<branch_name>`** for a hotfix, and **`service/<branch_name>`** for a microservice, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [ ✓] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [✓ ] There are no merge conflicts.
- [✓ ] There are no console warnings and errors.
- [✓ ] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [ ✓] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

**Review by clubkemp — COMMENTED** [review]
You made the PR to merge into the staging branch, you will need to change to be merging into feature/profile-redesign. Thats our collective working branch where we can consolidate everything and test before merging into staging

https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request
image

File: client/src/components/Button/ThankButton.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+const primaryColor = theme?.colors?.royalBlue;
+const whiteColor = theme?.colors?.offWhite;
+
+const ThanksButton= styled.div`
+display: flex;
+flex-direction: row;
+justify-content: center;
+align-items: center;
+padding: 8px 38px;
```
Comment (clubkemp) [inline client/src/components/Button/ThankButton.js]: @linadinh2602 just pointed out on my code we should be using rems instead of px, I agree :) 

I guess the question would be what if the figma designs have px in them...maybe it's ok then??

File: client/src/components/Button/ThankButton.js [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        <ThanksButton>♡Thanks</ThanksButton>
+        </>
+    )
+}
+
+export default ThankButton
+
+
+
+// import { Modal, Button } from "antd";
```
Comment (clubkemp) [inline client/src/components/Button/ThankButton.js]: I think we can get rid of the commented out code, if its a relic we don't need it in here, and if we ever DO need to go back and reference it we can just get it from and an older version

**Review by clubkemp — COMMENTED** [review]
could you use throw up some screenshots of the product, or if there is functionality we need to see try using screencastify (or similar) to make a little video or GIF?

**Comment by aditilamkhede** [comment]
> https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request

good catch @clubkemp

File: client/src/components/Button/ThankButton.js (hunk 1 above)
Comment (aditilamkhede) [reply client/src/components/Button/ThankButton.js]: @clubkemp figma is based on px only, you to change from px to rem and tweak a little. but yes please change all px to rem. In very cases where rem might now work or px works better is fine.

File: client/src/components/Button/ThankButton.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import styled from "styled-components";
+import { mq, theme } from "constants/theme";
+
+const primaryColor = theme?.colors?.royalBlue;
```
Comment (aditilamkhede) [inline client/src/components/Button/ThankButton.js]: here why condition?

File: client/src/components/Button/ThankButton.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+
+@media screen and (max-width: ${mq.phone.wide.maxWidth}) {
+    display: none;
+  }
+ `
+
+const ThankButton = () => {
+    return(
+        <>
+        <ThanksButton>♡Thanks</ThanksButton>
```
Comment (aditilamkhede) [inline client/src/components/Button/ThankButton.js]: is this an SVG from Figma?? We need to get SVG from figma. Check in existing icons inside src/assets/icons.

File: client/src/components/Button/ThankButton.js (hunk 2 above)
Comment (aditilamkhede) [reply client/src/components/Button/ThankButton.js]: Yes get rid of commented code and any extra white spaces/lines.

File: client/src/components/Feed/MessagesModal/MessageModal.js [hunk 5]
```diff
@@ -15,6 +15,7 @@ import { WebSocketContext } from "context/WebsocketContext";
 import { useTranslation } from "react-i18next";
 import TagManager from "react-gtm-module";
 import GTM from "constants/gtm-tags";
+import { MailruShareButton } from "react-share";
```
Comment (aditilamkhede) [inline client/src/components/Feed/MessagesModal/MessageModal.js]: are we suing this?? if not remove it.

File: client/src/components/Feed/MessagesModal/MessageModal.js [hunk 6]
```diff
@@ -161,7 +171,7 @@ const MessageModal = ({
             isFromUserCard={isFromUserCard}
             id={gtmId}
           >
-            <img src={activeemail} alt={"message-icon"} />
+            <img src={activeemail} className='envelopeIcon' alt={"message-icon"} />
```
Comment (aditilamkhede) [inline client/src/components/Feed/MessagesModal/MessageModal.js]: why this change?

File: client/src/components/Feed/MessagesModal/MessageModal.js [hunk 7]
```diff
@@ -71,6 +72,15 @@ const MessageModal = ({
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  // const envelopeColor = () => {
```
Comment (aditilamkhede) [inline client/src/components/Feed/MessagesModal/MessageModal.js]: remove commented code

File: client/src/components/Feed/MessagesModal/MessagesContainer.js [hunk 8]
```diff
@@ -3,6 +3,7 @@ import styled from "styled-components";
 import { mq, theme } from "constants/theme";
 
 const primaryColor = theme?.colors?.royalBlue;
```
Comment (aditilamkhede) [inline client/src/components/Feed/MessagesModal/MessagesContainer.js:5]: why conditional?

File: client/src/components/Picture/ProfilePic.js [hunk 9]
```diff
@@ -9,12 +9,14 @@ const InitialDiv = styled.div`
   margin: auto;
   margin-bottom: 1rem;
   border-radius: 50%;
-  border: ${(props) => (props.hasPhoto ? "none" : "0.2rem solid #425af2;")}
+  border: none;
+  border: ${(props) => (props.hasPhoto ? "none" : "none")}
```
Comment (aditilamkhede) [inline client/src/components/Picture/ProfilePic.js]: you have a photo or not border is none right? so do we need this line?

File: client/src/components/Profile/ProfileComponents.js [hunk 10]
```diff
@@ -116,13 +116,28 @@ export const NamePara = styled.p`
   overflow-wrap: break-word;
   margin: 0 !important;
   text-align: left;
-  max-width: 90%;
+  max-width: 70%;
   margin-top: 1rem;
+  font-family: Poppins;
+  font-weight: bold;
+  font-style: normal;
+  font-size: 26px;
```
Comment (aditilamkhede) [inline client/src/components/Profile/ProfileComponents.js]: change all px to rem and serach your color # in the project, if it already exists just use the color variable name

File: client/src/components/Profile/ProfileComponents.js [hunk 11]
```diff
@@ -210,14 +225,17 @@ export const HelpContainer = styled.div`
 export const UserInfoContainer = styled.div`
   background-color: ${colors.white};
   box-shadow: 0px 0.2rem 2rem rgba(0, 0, 0, 0.01);
-  max-width: 80rem;
+  max-width: 1173px;
```
Comment (aditilamkhede) [inline client/src/components/Profile/ProfileComponents.js]: change to rem

File: client/src/pages/Profile.js [hunk 12]
```diff
@@ -384,21 +390,33 @@ const Profile = ({
...
                 />
+                {/* <ThankButton>Thanks</ThankButton> */}
+
+                </>
               )}
             </NameDiv>
             {about && <DescriptionDesktop> {about} </DescriptionDesktop>}
             <IconsContainer>
-              <HelpContainer>
+              <MobileLocation>
```
Comment (aditilamkhede) [inline client/src/pages/Profile.js]: did you confirm with cami regarding location of address on mobile and desktop?

**Review by aditilamkhede — CHANGES_REQUESTED** [review]
@aleksandra-b-t , I added my comments. Also I see the profile as this now: 
Screen Shot 2021-02-25 at 6 55 42 PM

My concern is: 

1. Thank you button is not visible? 
2. Also, Why do you have to make changes in existing message container?

File: client/src/components/Feed/MessagesModal/MessagesContainer.js [hunk 13]
```diff
@@ -247,34 +270,35 @@ export const PrivateMessageContainer = styled.div`
   ${(props) =>
     props.isFromUserCard
       ? `
-    margin-left: 7.4rem;
+    // margin-left: 7.4rem;
```
Comment (linadinh2602) [inline client/src/components/Feed/MessagesModal/MessagesContainer.js]: this is a comment too and cause confusion

**Review by linadinh2602 — CHANGES_REQUESTED** [review]
I agree with Aditi that there are some syntax just need to make sure it clean

File: client/src/components/Button/ThankButton.js (hunk 3 above)
Comment (aleksandra-b-t) [reply client/src/components/Button/ThankButton.js]: this is how was used the main blue color in other files, which makes it easier for me to use name instead of #XXXX, is that ok?

File: client/src/components/Button/ThankButton.js (hunk 1 above)
Comment (aleksandra-b-t) [reply client/src/components/Button/ThankButton.js]: ok!

File: client/src/components/Button/ThankButton.js (hunk 4 above)
Comment (aleksandra-b-t) [reply client/src/components/Button/ThankButton.js]: This isn’t figma's svg this button isn’t done

File: client/src/components/Button/ThankButton.js (hunk 2 above)
Comment (aleksandra-b-t) [reply client/src/components/Button/ThankButton.js]: ok!

File: client/src/components/Feed/MessagesModal/MessageModal.js (hunk 5 above)
Comment (aleksandra-b-t) [reply client/src/components/Feed/MessagesModal/MessageModal.js]: interesting, I didn’t add id, should I remove it?

File: client/src/components/Feed/MessagesModal/MessageModal.js (hunk 7 above)
Comment (aleksandra-b-t) [reply client/src/components/Feed/MessagesModal/MessageModal.js]: I am still working on conditional rendering (white/blue) envelope depending on the device

File: client/src/components/Feed/MessagesModal/MessageModal.js (hunk 6 above)
Comment (aleksandra-b-t) [reply client/src/components/Feed/MessagesModal/MessageModal.js]: I had difficulty overwriting some general styles for envelope SVG. I needed to add className to specify which SVG I want to change.

File: client/src/components/Feed/MessagesModal/MessagesContainer.js (hunk 13 above)
Comment (aleksandra-b-t) [reply client/src/components/Feed/MessagesModal/MessagesContainer.js]: I will clean it!

File: client/src/components/Picture/ProfilePic.js (hunk 9 above)
Comment (aleksandra-b-t) [reply client/src/components/Picture/ProfilePic.js]: Yes, I will clean it

File: client/src/components/Profile/ProfileComponents.js (hunk 10 above)
Comment (aleksandra-b-t) [reply client/src/components/Profile/ProfileComponents.js]: Ok

File: client/src/components/Profile/ProfileComponents.js (hunk 11 above)
Comment (aleksandra-b-t) [reply client/src/components/Profile/ProfileComponents.js]: ok

File: client/src/pages/Profile.js (hunk 12 above)
Comment (aleksandra-b-t) [reply client/src/pages/Profile.js]: yes, confirmed

File: client/src/components/Button/ThankButton.js (hunk 3 above)
Comment (aditilamkhede) [reply client/src/components/Button/ThankButton.js]: yes, I meant remove conditional operator.

File: client/src/components/Picture/ProfilePic.js [hunk 14]
```diff
@@ -24,11 +25,14 @@ const InitialDiv = styled.div`
   @media screen and (min-width: ${(props) =>
     props.resolution ? props.resolution : mq.tablet.narrow.minWidth}) {
     margin: 0;
-    height: ${(props) => (props.hasPhoto ? `12.2rem` : `12rem`)};
+    height: ${(props) => (props.hasPhoto ? `137px` : `12rem`)};
```
Comment (aditilamkhede) [inline client/src/components/Picture/ProfilePic.js]: rem?

**Review by aditilamkhede — CHANGES_REQUESTED** [review]
I couldn't check the view as build url is throwing 505. But few minor changes like some places you are still using ox, if that can be changed to rem.

File: client/src/components/Feed/MessagesModal/MessagesContainer.js [hunk 15]
```diff
@@ -230,51 +246,72 @@ export const PrivateMessageContainer = styled.div`
...
-
   ${(props) =>
     props.isFromProfile
       ? `
-      display: flex;
-      align-items: center;
-      border: 0.1rem solid ${theme.colors.royalBlue};
+      position: absolute;
+      top: -3rem;
+      height: 6rem;
```
Comment (clubkemp) [inline client/src/components/Feed/MessagesModal/MessagesContainer.js:294]: I still think this is a little big. If I use the original figma numbers of 40px height and 152px width the button is smaller than how it appears in your design.  I used an online converted and did some tweaking and if you use root font-size 10 the conversion is more like 4rem for height, and 15.2rem for width

https://nekocalc.com/px-to-rem-converter

**Review by clubkemp — COMMENTED** [review]
Noticed when I was logged in I didn't see the edit profile as a button on desktop, only the pencil icon.

**Comment by SinghRomika** [comment]
@aleksandra-b-t  The build link given 
 https://featureprofile-redesign-user-profile-car.fightpandemics.xyz.
This seems to be the link for Fight pandemics user profile not for Doing Gud. The PRD in notion shows the design for Doin Gud not Fight Pandemics. Can you share the Figma designs for this?

**Comment by SinghRomika** [comment]
**Date: 03/07/21**
@aleksandra-b-t  Following is the verification details of the build link : 
https://featureprofile-redesign-user-profile-car.fightpandemics.xyz. 
**DESKTOP VERSION**
Version:  Chrome 89 Beta, IOS 14
OS: Windows10, MacOS BigSur
Browsers: Chrome, Safari, Edge
Resolutions: 1920 * 1080
**According to my observation following is working as expected** 
- Background image can stay blue like it currently is instead of the ocean background.
- Profile's photo have white border.
- Location is under name.
- 'Own profile' view in mobile and desktop version stays like in old version (include edit icon).

**According to my observation following is not working as expected** 
-  **Thanks button should be hidden under the message icon according to the PRD**
Thanks button should be hidden under the message icon according to the PRD

- The Background image is not as per the designs in Figma, the edge on the left side is not curved as the edge of the background image on right, they should have symmetry and consistency. I don't know if this issues is relevant to PR#2241 or not, so plz feel free to comment on this one if its not?

Background image should have symmetry

**Comment by SinghRomika** [comment]
**Date: 03/07/21**
@aleksandra-b-t Following is the verification details of the build link :
https://featureprofile-redesign-user-profile-car.fightpandemics.xyz.
**MOBILE VERSION:**
Brands: Samsung, Huawei, iPhone,  iPad.
Device: Galaxy(A71 5G, Note 9, Note 20, Note 10) Huawei P30 Pro, iPhone(12 Pro Max, 12 Pro, 8th Gen)
Browser: Chrome, Samsung Internet, Safari, Chromium
OS: Android, IOS 14.4

**According to my observation following is working as expected**

- The location icon can be a blue circle for now like the web design below
- The location is under description.
- Profile photo doesn’t have border.
- 'Own profile' view in mobile and desktop version stays like in old version (include edit icon)
- 'Own profile' view in mobile version doesn’t include 'envelope button'.


**According to my observation following is not working as expected**
- The Location icon is too close to the Location text, according to the Figma design it should not be that close
location icon

**Comment by SinghRomika** [comment]
DATE: 03/07/21
@aleksandra-b-t  Is the Navigation Bar ready for this feature? 
According to the PRD the 
-Navigation bar in mobile version stays like in old versions for consistency of view. CAN YOU PLZ COMMENT ON THIS?

**Comment by aditilamkhede** [comment]
> * **Thanks button should be hidden under the message icon according to the PRD**
@SinghRomika I didn't get this. This is not clear. Button should be hidden and it is not there right? So, what's the issue?

**Comment by aditilamkhede** [comment]
> * The Background image is not as per the designs in Figma, the edge on the left side is not curved as the edge of the background image on right, they should have symmetry and consistency. I don't know if this issues is relevant to PR#2241 or not, so plz feel free to comment on this one if its not?

@SinghRomika For this, I asked Cami and she said we can ignore that for now.

**Comment by aditilamkhede** [comment]
> DATE: 03/07/21
> @aleksandra-b-t Is the Navigation Bar ready for this feature?
> According to the PRD the
> -Navigation bar in mobile version stays like in old versions for consistency of view. CAN YOU PLZ COMMENT ON THIS?

@SinghRomika Could you please elaborate on this? Which Navigation Bar?

**Comment by SinghRomika** [comment]
> > * **Thanks button should be hidden under the message icon according to the PRD**
> >   @SinghRomika I didn't get this. This is not clear. Button should be hidden and it is not there right? So, what's the issue?

Date: 03/08/21
@aditilamkhede Seems like the Thanks Button is not showing up since the THANKS feature is still under development and it is a future feature.

**Comment by SinghRomika** [comment]
> > DATE: 03/07/21
> > @aleksandra-b-t Is the Navigation Bar ready for this feature?
> > According to the PRD the
> > -Navigation bar in mobile version stays like in old versions for consistency of view. CAN YOU PLZ COMMENT ON THIS?
> 
> @SinghRomika Could you please elaborate on this? Which Navigation Bar?

Date: 03/08/21
@aditilamkhede I wasn't sure about which navigation bar i am suppose to test? I confirmed with @aleksandra-b-t  and its this one 
Navigation Bar

**Comment by SinghRomika** [comment]
@aleksandra-b-t Thanks for showing the navigation bar. I have verified the Navigation bar in mobile version, it looks as expected.
 The Navigation bar in mobile version stays like in old versions for consistency of view.

**Comment by SinghRomika** [comment]
> > * The Background image is not as per the designs in Figma, the edge on the left side is not curved as the edge of the background image on right, they should have symmetry and consistency. I don't know if this issues is relevant to PR#2241 or not, so plz feel free to comment on this one if its not?
> 
> @SinghRomika For this, I asked Cami and she said we can ignore that for now.

Date:03/08/21
@aditilamkhede  Ok!

**Comment by SinghRomika** [comment]
Thanks!@aleksandra-b-t @aditilamkhede 
 The issue persisting in the **MOBILE VERSION** earlier seems to be working as expected  according to my observations made on multiple Mobile devices:
OS: Android(11.0, 10.0, 9.0 ) IOS: 14.4
Device: Samsung( A71 5G, Note 20, Tab S7 ) Huawei P30 Pro, iPhone 12 Pro Max, iPad Air 4th Gen
Browser: Chrome, Safari.

The Location icon is according to the Figma design.
Location icon PR#2241

File: client/src/components/Picture/ProfilePic.js [hunk 16]
```diff
@@ -24,11 +25,14 @@ const InitialDiv = styled.div`
...
     line-height: ${(props) => (props.hasPhoto ? `10rem` : `11rem`)};
-    width: ${(props) => (props.hasPhoto ? `12.2rem` : `12rem`)};
+    width: ${(props) => (props.hasPhoto ? `137px` : `12rem`)};
```
Comment (aditilamkhede) [inline client/src/components/Picture/ProfilePic.js]: is it possible to change this to 
rem?

**Review by aditilamkhede — APPROVED** [review]
@aleksandra-b-t  LGTM, just some minor changes

**Comment by AndersonNY** [comment]
https://fightpandemics.slack.com/files/U01NUUC4MDM/F01R45343R9/image_from_ios.jpg  Tested on iPhone 11 and it seems working as expected.

File: client/src/components/Picture/ProfilePic.js (hunk 16 above)
Comment (aleksandra-b-t) [reply client/src/components/Picture/ProfilePic.js]: sure!

**Review by aditilamkhede — APPROVED** [review]
LGTM

**Review by clubkemp — APPROVED** [review]
Looing good!

**Comment by AndersonNY** [comment]
The Location icon is correct according to the Figma design. I checked the mobile version with, Samsun iPhone 12 Pro Max, iPhone XS Max, iPhone 6 plus, LG
Screen Shot 2021-03-15 at 12 03 41 PM
Screen Shot 2021-03-15 at 12 07 38 PM

**Comment by aleksandra-b-t** [comment]
feature ready

**Comment by AndersonNY** [comment]
Screen Shot 2021-03-17 at 10 35 02 PM
Screen Shot 2021-03-17 at 10 35 14 PM

Request  and offers are working as expected.
